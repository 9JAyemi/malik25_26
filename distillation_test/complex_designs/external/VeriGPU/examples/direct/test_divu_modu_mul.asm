; mul

li x1, 15
li x2, 31
mul x3, x1, x2
outr x3  ; 465

li x1, 0
li x2, 31
mul x3, x1, x2
outr x3  ; 0

li x1, 15
li x2, 0
mul x3, x1, x2
outr x3  ; 0

li x1, 347911
li x2, 12345
mul x3, x1, x2
outr x3  ; 4294961295

; divu, remu

li x1, 4
li x2, 4
divu x3, x1, x2
remu x4, x1, x2
outr x3  ; 1 r 0
outr x4

li x1, 4
li x2, 2
divu x3, x1, x2
remu x4, x1, x2
outr x3  ; 2 r 0
outr x4

li x1, 0
li x2, 0
divu x3, x1, x2
remu x4, x1, x2
outr x3  ; ffffffff r 0
outr x4

li x1, 0
li x2, 2
divu x3, x1, x2
remu x4, x1, x2
outr x3  ; 0 r 0
outr x4

li x1, 2222
li x2, 123
divu x3, x1, x2
remu x4, x1, x2
outr x3  ; 18 r 8
outr x4

li x1, 1111
li x2, 444
divu x3, x1, x2
remu x4, x1, x2
outr x3  ; 2 r 223
outr x4

halt
