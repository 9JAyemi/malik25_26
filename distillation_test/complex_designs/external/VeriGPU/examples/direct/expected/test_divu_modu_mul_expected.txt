out 0 00000000000000000000000111010001 000001d1 465
out 1 00000000000000000000000000000000 00000000 0
out 2 00000000000000000000000000000000 00000000 0
out 3 11111111111111111110100010001111 ffffe88f 4294961295
out 4 00000000000000000000000000000001 00000001 1
out 5 00000000000000000000000000000000 00000000 0
out 6 00000000000000000000000000000010 00000002 2
out 7 00000000000000000000000000000000 00000000 0
out 8 11111111111111111111111111111111 ffffffff 4294967295
out 9 00000000000000000000000000000000 00000000 0
out 10 00000000000000000000000000000000 00000000 0
out 11 00000000000000000000000000000000 00000000 0
out 12 00000000000000000000000000010010 00000012 18
out 13 00000000000000000000000000001000 00000008 8
out 14 00000000000000000000000000000010 00000002 2
out 15 00000000000000000000000011011111 000000df 223
